Wikipedia/UI-V5/WMFNearbyViewModel.m

```diff
@@ -0,0 +1,193 @@
+//
+//  WMFNearbyViewModel.m
+//  Wikipedia
+//
+//  Created by Brian Gerstle on 9/9/15.
+//  Copyright (c) 2015 Wikimedia Foundation. All rights reserved.
+//
+
+#import "WMFNearbyViewModel.h"
+
+// Location & Fetching
+#import "WMFLocationManager.h"
+#import "WMFLocationSearchFetcher.h"
+
+// Models
+#import "WMFLocationSearchResults.h"
+#import "MWKLocationSearchResult.h"
+#import "MWKHistoryEntry.h"
+#import "CLLocation+WMFApproximateEquality.h"
+#import "CLLocation+WMFBearing.h"
+
+// Frameworks
+#import "Wikipedia-Swift.h"
+#import "PromiseKit.h"
+
+@interface WMFNearbyViewModel ()
+<WMFNearbyControllerDelegate>
+
+@property (nonatomic, strong) WMFLocationSearchFetcher* locationSearchFetcher;
+@property (nonatomic, strong) WMFLocationManager* locationManager;
+@property (nonatomic, assign) NSUInteger resultLimit;
+
+@property (nonatomic, strong, readwrite) WMFLocationSearchResults* locationSearchResults;
+
+@property (nonatomic, strong) NSMutableIndexSet* autoUpdatingIndexes;
```

Comment:
So what does adding the complexity of tracking specific indexes buy us?

It seems like its only needed because we moved the user distance and bearing calculations into this class instead of closer to the view where they are used.

Before… we didn't have to worry because we only updated the cells that are on screen… but now since the logic is deeper… we have to add complexity to track what should be updated… and the logic is "eager" instead of called on demand like before (just update this particular cell please)

So to me it just looks like some of the refactoring is making tradeoffs that I am not sure are better than before…

1. We put the user distance and bearing strings on the search results, making them mutable…
2. So we can set them in this object…
3. Which is too far away from the view to be smart about which objects it should be updating (doesn't know which cells are on screen)…
4. So we add the complexity of tracking indexes.

I'm not sure this is a good trade off. It seems like this class really wants to do some things, but isn't in the right part of the object graph to do them. If this object were not a view model… and were permitted to update the cells… you could avoid some of this complication.